core/net/rpl/rpl-icmp6.c

```diff
@@ -887,6 +887,7 @@ dao_input_nonstoring(void)
   }
 
   if(flags & RPL_DAO_K_FLAG) {
+    uip_clear_buf();
     dao_ack_output(instance, &dao_sender_addr, sequence);
```

Comment:
I thought you would add uip_clear_buf() there by yourself; but you might want me to do so....

I've added two uip_clear_buf()s to make sure to clear the uIP buffer before calling dao_ack_output().